internal/router/controllers/user.go

```diff
@@ -359,5 +269,69 @@ func (c *UserController) runReadRequest(
 		_ = json.NewEncoder(rw).Encode(errors.ServiceError{Message: "Row not found"})
 		return nil, false
 	}
-	return rows, true
+	return res, true
+}
+
+func formatterOptions(r *http.Request) ([]formatter.FormatOption, error) {
+	var opts []formatter.FormatOption
+	params, err := getFormatterParams(r)
+	if err != nil {
+		return nil, err
+	}
+	if params.output != nil {
+		opts = append(opts, formatter.WithOutput(*params.output))
+	}
+	if params.extract != nil {
+		opts = append(opts, formatter.WithExtract(*params.extract))
+	}
+	if params.unwrap != nil {
+		opts = append(opts, formatter.WithUnwrap(*params.unwrap))
+	}
+	return opts, nil
+}
+
+type formatterParams struct {
+	output  *formatter.Output
+	extract *bool
+	unwrap  *bool
+}
+
+func getFormatterParams(r *http.Request) (formatterParams, error) {
+	c := formatterParams{}
+	output := r.URL.Query().Get("output")
+	extract := r.URL.Query().Get("extract")
+	unwrap := r.URL.Query().Get("unwrap")
+	if output != "" {
+		output, ok := formatter.OutputFromString(output)
+		if !ok {
+			return formatterParams{}, fmt.Errorf("bad output query parameter")
+		}
+		c.output = &output
+	}
+	if extract != "" {
+		extract, err := strconv.ParseBool(extract)
+		if err != nil {
+			return formatterParams{}, err
+		}
+		c.extract = &extract
+	}
+	if unwrap != "" {
+		unwrap, err := strconv.ParseBool(unwrap)
+		if err != nil {
+			return formatterParams{}, err
+		}
+		c.unwrap = &unwrap
+	}
+
+	// Special handling for old mode param
+	mode := r.URL.Query().Get("mode")
+	if mode == "list" {
+		v := true
+		c.unwrap = &v
+	} else if mode == "json" {
+		v := formatter.Objects
+		c.output = &v
+	}
+
+	return c, nil
```

Comment:
This is just a helper to convert untypes request parameters into something typed. Note the support for legacy `mode` param. We can keep this around for a while in case some smart contracts or devs are using it.

Reply:
I just chose some reasonable legacy things to support because they easily map to the new flags/behavior. Things like `csv` mode are dropped.